### PR TITLE
Cb/fix ybus

### LINKED ISSUE #253: PowerSystems RTS-GMLC parsing error for PTDF matrix
In PowerSystems when I build the RTS system with a ptdf matrix the function
buildptdf throws an error at this line 45
slacks = [num_bus[n.number] for n in nodes if n.bustype == "SF"]
The reason is that there is no bus in the RTS of type SF
There is a bus of type Ref
If I change the line of code to be
slacks = [num_bus[n.number] for n in nodes if n.bustype == "Ref"]
the code runs for the RTS.
Is this the correct fix or is the Ref bus in the RTS different then the Slack bus SF for the system?
I think the RTS is parsing the slack bus as type Ref when it should parse it as type SF

### PR DESCRIPTION
**Pull request by claytonpbarrows**
addresses #256 

 - [x] needs a Ybus5nonsequential_matpower for testing
 - [x] do the same for LODF

closes #253

### CODE REVIEW

File: src/utils/network_calculations/ybus_calculations.jl [hunk 1]
```diff
@@ -0,0 +1,200 @@
+
+struct Ybus <: PowerNetworkMatrix
+    data::SparseArrays.SparseMatrixCSC{Complex{Float64},Int64}
+    axes::NTuple{2,Array}
+    lookup::NTuple{2,Dict}
+end
+
+function ybus!(ybus::SparseArrays.SparseMatrixCSC{Complex{Float64},Int64},
```
Comment (jd-lara) [inline src/utils/network_calculations/ybus_calculations.jl]: make `_ybus!`

File: src/utils/network_calculations/ybus_calculations.jl [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    ybus[b.transformer.connectionpoints.to.number,
+        b.transformer.connectionpoints.from.number] += Y12;
+    Y22 = (y_a + y * (1 - c)) ;
+    ybus[b.transformer.connectionpoints.to.number,
+        b.transformer.connectionpoints.to.number] += Y22;
+
+end
+=#
+
+# Old Ybus creation
```
Comment (jd-lara) [inline src/utils/network_calculations/ybus_calculations.jl]: We can delete this

File: src/utils/network_calculations/ybus_calculations.jl [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+end
+
+function Ybus(branches::Array{T}, nodes::Array{Bus}) where {T<:ACBranch}
+
+    #Get axis names
+    bus_ax = [get_number(bus) for bus in nodes]
+    axes = (bus_ax, bus_ax)
+    look_up = (_make_ax_ref(bus_ax),_make_ax_ref(bus_ax))
+
+    ybus, A = _buildybus(branches, nodes)
```
Comment (jd-lara) [inline src/utils/network_calculations/ybus_calculations.jl]: We don't need to return A here

File: src/utils/network_calculations/ptdf_calculations.jl [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    elseif length(slack_position) == 0
+        @warn("Slack bus not identified in the Bus/Nodes list, can't build PTDF")
+        S = Array{Float64,2}(undef,linecount,buscount)
+    end
+
+    return  S, A
+
+end
+
+function PTDF(branches::Array{T}, nodes::Array{Bus}, dist_slack::Array{Float64}=[0.1]) where {T<:ACBranch}
```
Comment (jd-lara) [inline src/utils/network_calculations/ptdf_calculations.jl]: We should add a `PTDF(sys::System)` so we don't need to get the branches and buses to make the PTDF

File: src/utils/network_calculations/ybus_calculations.jl [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+        ybus!(ybus, b, num_bus)
+
+
+    end
+
+    return  ybus, A
+
+end
+
+function Ybus(branches::Array{T}, nodes::Array{Bus}) where {T<:ACBranch}
```
Comment (jd-lara) [inline src/utils/network_calculations/ybus_calculations.jl]: We can also make a `Ybus(sys::System)`

**Comment by claytonpbarrows** [comment]
@jd-lara the tests pass on this PR locally, but Travis is broken due to the powersystemsdata changes.

**Comment by jd-lara** [comment]
@claytonpbarrows you need to do a rebase to address the conflicts as well.

**Pull request merged**